test/decode-events.test.ts

```diff
@@ -0,0 +1,65 @@
+import { expect } from "chai";
+import { starknet } from "hardhat";
+import { StarknetContract, StarknetContractFactory } from "hardhat/types/runtime";
+import { TIMEOUT } from "./constants";
+import { BigNumber } from "ethers";
+import { expectFeeEstimationStructure } from "./util";
+
+/**
+ * Receives a hex address, converts it to bigint, converts it back to hex.
+ * This is done to strip leading zeros.
+ * @param address a hex string representation of an address
+ * @returns an adapted hex string representation of the address
+ */
+ function adaptAddress(address: string) {
+  return "0x" + BigInt(address).toString(16);
+}
+
+const OK_TX_STATUSES = ["PENDING", "ACCEPTED_ON_L2", "ACCEPTED_ON_L1"]
```

Comment:
Move this and `adaptAddress` to `util.ts`. Also modify `sample-test.ts` to import from there.